include/lsst/afw/image/fits/fits_io.h

```diff
@@ -56,8 +56,14 @@ inline void fits_read_array(fits::Fits& fitsfile, ndarray::Array<PixelT, 2, 2>&
         }
         shape = shape3.last<2>();
     }
-
-    fitsfile.readMetadata(metadata, true);
+    //
+    // Note that we cannot call fitsfile.readMetadata(metadata, true)
+    // as it doesn't handle INHERIT correctly (DM-11235)
+    //
+    const int hdu = fitsfile.getHdu();
+    std::shared_ptr<daf::base::PropertyList> header(readMetadata(fitsfile, true));
```

Comment:
It won't compile; I think that the problem is that there's some PropertyList copy/move operator that doesn't exist.